squash_iterator.go

```diff
@@ -552,6 +552,109 @@ func (i *squashRefIter) Schema() sql.Schema {
 	return RefsSchema
 }
 
+type squashRefIndexIter struct {
+	ctx           *sql.Context
+	index         sql.IndexLookup
+	iter          sql.RowIter
+	filters       sql.Expression
+	pool          *RepositoryPool
+	ref           *Ref
+	repo          *Repository
+	row           sql.Row
+	skipGitErrors bool
+}
+
+// NewIndexRefsIter returns an iterator that will return all references
+// that match the given filters in the given index.
+func NewIndexRefsIter(filters sql.Expression, index sql.IndexLookup) RefsIter {
+	return &squashRefIndexIter{filters: filters, index: index}
+}
+
+func (i *squashRefIndexIter) Repository() *Repository { return i.repo }
+func (i *squashRefIndexIter) Ref() *Ref               { return i.ref }
+func (i *squashRefIndexIter) Close() error {
+	return i.iter.Close()
+}
+func (i *squashRefIndexIter) New(ctx *sql.Context, pool *RepositoryPool) (ChainableIter, error) {
```

Comment:
maybe we can rename it as `NewChainableIter` - wdyt?

Reply:
This is the name in the `ChainableIter` interface